### ISSUE TITLE
Allow one line block

### ISSUE THREAD

**Issue by ofirule** [hashicorp/terraform#22086]
version : Terraform v0.12.4

I think using the following block syntax might be easier in some use cases. 
See example:

```
resource "aws_security_group" "backend_sg" {
  name = local.sg_name
  vpc_id = aws_vpc.backend_vpc.id
  ingress { protocol = "tcp" from_port = 22    to_port = 22    cidr_blocks = ["0.0.0.0/0"] }
  ingress { protocol = "tcp" from_port = 80    to_port = 80    cidr_blocks = ["0.0.0.0/0"] ipv6_cidr_blocks ["::/0"] }
  ingress { protocol = "tcp" from_port = 443   to_port = 443   cidr_blocks = ["0.0.0.0/0"] ipv6_cidr_blocks ["::/0"] }


  // Terraform removes the default rule
  egress  { protocol = "-1"  from_port = 0  to_port = 0  cidr_blocks = ["0.0.0.0/0"] }
}
```

This is the error I got:
```
Error: Invalid single-argument block definition

  on ../../modules/backend/backend.tf line 56, in resource "aws_security_group" "backend_sg":
  56:   ingress { protocol = "tcp" from_port = 22    to_port = 22    cidr_blocks = ["0.0.0.0/0"] }

A single-line block definition must end with a closing brace immediately after
its single argument definition.
```

**Comment by SPhu** [comment]
I don't know why this was take away but I use it quite frequently in my code for things like tags and it improves the readability of the files.  I would like to see it return it possible.

**Comment by jasonmoo** [comment]
This is a requirement for some of the larger configuration files I have to maintain readability.  Please stop breaking things.

**Comment by acutchin-bitpusher** [comment]
Agree. Tabular layouts are much easier to read and quickly understand than multi-line layouts.

**Comment by chenrui333** [comment]
This is really annoying, is there anyway to bypass this issue.

By the way, `terraform fmt` does work on the resource file.

**Comment by techdragon** [comment]
When defining a lot of environment blocks for kubernetes resources, it gets very annoying to have short statements forced out over multiple lines like this.
```hcl
env { name = "INSTANCE_IP", value_from = { field_ref = { field_path = "status.hostIP" } } }
```
and 
```hcl
http_get { path = "/watchman/", scheme = "HTTP", port = "5000" }
```
Thats how I would like to write them, comma separated, so as to be completely explicit and unambiguous. However[ this code here ](https://github.com/hashicorp/hcl/blob/hcl2/hclsyntax/parser.go#L362-L385 ) is why I have to write that out as 
```hcl
env {
  name = "INSTANCE_IP"
  value_from = { field_ref = { field_path = "status.hostIP" } } 
}
```
and 
```hcl 
http_get { 
  path = "/watchman/", 
  scheme = "HTTP", 
  port = "5000" 
}
```

Which when multiplied dozens of times over a single resource block, can turn a 16-32 line terraform file, into a 64 to 256 line file _extremely easily_ which is just downright frustrating.

**Comment by aep** [comment]
is this a wontfix?

**Comment by maksym-petrov-ct** [comment]
still waiting. 
It shouldn't be too hard to tweak the internal terraform parser ?

**Comment by giner** [comment]
In the recent versions (1.5.x) `terraform fmt` complains with:
> A single-line block definition can contain only a single argument. If you meant to define argument "secret_key_ref", use an equals sign to assign it a value. To define a nested block, place it on a line of its own within its parent block.

**Comment by giner** [comment]
This works:
```
selector { match_labels = { app = "keycloak" } }
```

This does not:
```
env { name = "KEYCLOAK_ADMIN", value = "superadm" } 
```

**Comment by r12f** [comment]
can we have a flag or something to disable this error?

**Comment by crw** [comment]
Per the design of HCL 2.0, it is valid to use a one-line declaration only if there are zero or one arguments inside a block. This was an intentional design limitation and most likely will not be changed in the future (at least not in HCL 2.0). 

@r12f please open a new issue for the request (if you feel strongly about it).
